PowerUp/app/src/main/java/powerup/systers/com/GameActivity.java

```diff
@@ -81,86 +85,47 @@ public void onCreate(Bundle savedInstanceState) {
         questionTextView.setMovementMethod(new ScrollingMovementMethod());
 
-        // set eye,skin,cloth,hair,accessory to avatar
-        String eyeImageName = getResources().getString(R.string.eye);
-        eyeImageName = eyeImageName + getmDbHandler().getAvatarEye();
-        R.drawable ourRID = new R.drawable();
-        java.lang.reflect.Field photoNameField;
-        try {
-            photoNameField = ourRID.getClass().getField(eyeImageName);
-            eyeImageView.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException error) {
-            error.printStackTrace();
-        }
-
-        String skinImageName = getResources().getString(R.string.skin);
-        skinImageName = skinImageName + getmDbHandler().getAvatarSkin();
-        try {
-            photoNameField = ourRID.getClass().getField(skinImageName);
-            skinImageView.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException error) {
-            error.printStackTrace();
-        }
-
-        String clothImageName = getResources().getString(R.string.cloth);
-        clothImageName = clothImageName + getmDbHandler().getAvatarCloth();
-        try {
-            photoNameField = ourRID.getClass().getField(clothImageName);
-            clothImageView.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException error) {
-            error.printStackTrace();
-        }
-
-        String hairImageName = getResources().getString(R.string.hair);
-        hairImageName = hairImageName + getmDbHandler().getAvatarHair();
-        try {
-            photoNameField = ourRID.getClass().getField(hairImageName);
-            hairImageView.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException error) {
-            error.printStackTrace();
-        }
-
-        getmDbHandler().setAvatarAccessory(getmDbHandler().getAvatarAccessory());
-        String accessoryImageName = getResources().getString(R.string.accessories);
-        accessoryImageName = accessoryImageName + getmDbHandler().getAvatarAccessory();
-        try {
-            photoNameField = ourRID.getClass().getField(accessoryImageName);
-            accessoryImageView.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException error) {
-            error.printStackTrace();
-        }
+        // use DbResourceUtil to extract data from database & apply it on avatar
+        DbResourceUtils resourceUtils = new DbResourceUtils(dataSource, context);
+        eyeImageView.setImageResource(resourceUtils.getEyeResourceId());
```

Comment:
Do you think, any of this view could be null, and you should check for null and before calling setImageResource on each of view? if any of this is null, its good idea to log appropriate message.